### ISSUE TITLE
Use ssh key inside a devcontainer

### ISSUE THREAD

**Issue by bilby91** [coder/coder#7508]
Hello,

I would like to start by just saying that coder is awesome ❤️!

I'm currently implementing Coder for our internal projects at Crunchloop. We are starting to use Coder to deploy VMs using fly.io. The coder workspaces are really simple for us in terms of provisioning since we literally create them so that we can spin up devcontainers.

At the moment we have a configuration that is working but there are some details that we want to improve. One of those things that is bugging us is the ability to reuse the ssh key that Coder "injects" for us in the workspace. Since we work at the devcontainer layer, we can't use the `GIT_SSH_COMMAND` that is configured in fly.io VM (or at least that is my understanding). Coder doesn't seem to deploy a standard SSH Agent where we can do Agent Forwarding. If we would be able to have a standard SSH Agent running, we would be able to connect to the Dev Containers using ssh and enable forwarding. I've been doing this in the past in different linux hosts when running devcontainers.

So, I'm wondering if there is any workaround that we can do in order to inject/proxy the private key to the devcontainers or expose them in some kind of fashion.

Looking forward for any feedback!

**Comment by bilby91** [comment]
I was able to get this working in a hacky way.

Snippet of the relevant pieces of my docker-compose file used by the devcontainer in case someone finds it useful.

```yaml
version: "3.8"

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile

    volumes:
      # `GIT_ASKPASS` holds a pointer to the coder binary that is used
      # by `GIT_SSH_COMMAND` to interact with coder's private ssh key.
      - $GIT_ASKPASS:$GIT_ASKPASS

    environment:
      # Replace `ssh` with coder's implementation
      #
      - GIT_SSH_COMMAND=$GIT_SSH_COMMAND

      # We need the following varibales to be able to invoke `GIT_SSH_COMMAND`
      # successfully
      #
      - CODER_AGENT_AUTH=$CODER_AGENT_AUTH
      - CODER_AGENT_TOKEN=$CODER_AGENT_TOKEN
      - CODER_AGENT_URL=$CODER_AGENT_URL
```

**Comment by matifali** [comment]
Do you have any specific reason for using a dev container inside the VM? Coder allows you to create templates per use case or a single template with multiple images.
Some details in this blog post: https://coder.com/blog/managing-templates-in-coder
And if you also need to deploy coder on fly.io: https://coder.com/blog/remote-developer-environments-on-fly-io

**Comment by matifali** [comment]
BTW, Coder has dev containers native support on roadmap.
See: https://coder.com/roadmap

**Comment by bilby91** [comment]
@matifali Yep, I've seen that item in the roadmap.

While experimenting with all these stuff I did something very similar to this => https://github.com/bpmct/coder-templates/blob/1287f9acac0635e8b725d0eadd59d3daabc600a3/gcp-devcontainer/main.tf#L110-L132

I managed to get it working (the only hack was that the dev container specification had to declare that `CODER_AGENT_TOKEN` need to be forwarded to the dev container since the agent is running inside it).

Even though everything worked, I didn't like the experience because VS Code is not aware that is running in a dev container and constantly ask the user if he wants to spin up the dev containers).

I'm really curious about what native support would mean. I would love being able to collaborate or try things as they are getting worked out!
